apps/taxes/sentry.client.config.ts

```diff
@@ -1,37 +1,35 @@
 /*
- * This file configures the initialization of Sentry on the client.
- * The config you add here will be used whenever a users loads a page in their browser.
+ * This file configures the initialization of Sentry on the browser.
+ * The config you add here will be used whenever a page is visited.
  * https://docs.sentry.io/platforms/javascript/guides/nextjs/
  */
 
 import * as Sentry from "@sentry/nextjs";
-import pkg from "./package.json";
+import { BaseError } from "./src/error";
+import { shouldExceptionLevelBeReported } from "./src/sentry-utils";
 
-Sentry.init({
-  dsn: process.env.NEXT_PUBLIC_SENTRY_DSN,
+const SENTRY_DSN = process.env.SENTRY_DSN || process.env.NEXT_PUBLIC_SENTRY_DSN;
+const SENTRY_ENVIRONMENT = process.env.SENTRY_ENVIRONMENT;
 
+Sentry.init({
+  dsn: SENTRY_DSN,
   // Adjust this value in production, or use tracesSampler for greater control
-  tracesSampleRate: 0.5,
-
-  // Setting this option to true will print useful information to the console while you're setting up Sentry.
-  debug: false,
+  tracesSampleRate: 1.0,
+  environment: SENTRY_ENVIRONMENT,
+  includeLocalVariables: true,
+  ignoreErrors: ["TRPCClientError"],
+  beforeSend(errorEvent, hint) {
+    const error = hint.originalException;
 
-  replaysOnErrorSampleRate: 1.0,
+    if (error instanceof BaseError) {
+      errorEvent.level = error.sentrySeverity;
 
-  /*
-   * This sets the sample rate to be 10%. You may want this to be 100% while
-   * in development and sample at a lower rate in production
-   */
-  replaysSessionSampleRate: 0.1,
+      // Ignore exceptions below specified severity (warning default)
+      if (!shouldExceptionLevelBeReported(errorEvent.level ?? "error")) {
+        return null;
+      }
+    }
 
-  // You can remove this option if you're not planning to use the Sentry Session Replay feature:
-  integrations: [
-    new Sentry.Replay({
-      // Additional Replay configuration goes in here, for example:
-      maskAllText: true,
-      blockAllMedia: true,
-    }),
-  ],
```

Comment:
Do we want to remove Sentry Replay? I know it was missing in the Payments App, but maybe we still want this feature?

Reply:
Is there a reason it was removed? I think its cool

Reply:
brought it back